iceoryx_utils/include/iceoryx_utils/posix_wrapper/pthread.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_UTILS_POSIX_WRAPPER_PTHREAD_HPP
+#define IOX_UTILS_POSIX_WRAPPER_PTHREAD_HPP
+
+#include "iceoryx_utils/cxx/expected.hpp"
+#include "iceoryx_utils/cxx/smart_c.hpp"
+#include "iceoryx_utils/cxx/string.hpp"
+
+#include <pthread.h>
+
+namespace iox
+{
+namespace posix
+{
+enum class PThreadErrorType : int32_t
+{
+    EXCEEDED_RANGE_LIMIT = 0
+};
+
+cxx::expected<PThreadErrorType> setThreadName(pthread_t thread, const char* name);
```

Comment:
Since this is a C++ wrapper you should create an alias like `using threadName_t = cxx::string<16>` and use this type instead of `const char*`. Like `cxx::expected<PThreadErrorType> setThreadName(pthread_t thread, const threadName_t & name);`

Reply:
I had exactly the same thought, a moment after pushing :smile: Done